test/unit_tests/cellular/cellular.cpp

```diff
@@ -22,12 +22,33 @@
 #include "net_hal.h"
 #include <limits>
 #include "spark_wiring_cellular_printable.h"
+#include "appender.h"
 
 #undef WARN
 #undef INFO
 #define CATCH_CONFIG_MAIN
 #include "catch2/catch.hpp"
 
+namespace {
+
+    class FakeStream: public Print {
+    private:
+        BufferAppender append_;
+
+    protected:
+        virtual size_t write(uint8_t c) override {
+            append_.append(&c, 1);
```

Comment:
I don't know enough about `BufferAppender::append`, why does this require a pointer?

Is it expecting a temporary buffer to copy the contents out of, or is it going to try and hold onto the pointer?

Reply:
It's just one of two defined overloads for the `BufferAppender::append` and I picked the first to be explicit.  The second will do the same thing but in the background.